Jumoo/Our.Umbraco.MaintenanceMode
# V10 load balanced

Resolves #33 

## Summary
This PR adds support for load balanced environments. It proposes the introduction of a few ideas: 
* An `IStorageProvider` in the service code to access the appropriate place to read/write settings
* An implementation of it - `DatabaseStorageProvider` for use in load balanced setups
* A custom notification to broadcast when settings change (useful for clearing cache or sending emails)

## Testing
How to test?
Add the following to appsettings: 
```json
  "MaintenanceMode": {
    "StorageMode": "Database"
  }
```
This will trigger the switch into database storage mode, observe that settings are written to and read from the database. 

Note that without the app setting above, the code will attempt to determine whether Umbraco is in a load balanced setup and behave accordingly. 

Also @KevinJump sorry it looks like a lot of files changed but there are a lot of one-liners in the notification handlers :)
